substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1748,6 +1808,23 @@ impl<T: Config> StakingInterface for Pallet<T> {
 			.map(|_| ())
 	}
 
+	fn update_payee(stash: &Self::AccountId, reward_acc: &Self::AccountId) -> DispatchResult {
```

Comment:
I'd prefer this `update_payee` method to take a `RewardDestination<AccountId>` instead of a reward account. That way we could allow the caller to set payee to `RewardDestination::None`, `RewardDestination::Staked`, etc or any other variants we have have in the future.

Reply:
In any case, it seems that the reward destination for virtual accounts should always be `Stash(accountid)`. But it would be helpful to have this trait as general as possible so that it can be used in any case.

Reply:
`RewardDestination` is an implementation detail, not just conceptually but also in the code (it's not part of staking primitives, probably for a reason). Looking at the rest of the interface, I don't think it fits naturally here and it wouldn't make sense to bring it over. If other (more general) use cases appear, I'd argue it's more ergonomic to add new functions serving them and the impls behind the scenes can use their own structures (like `RewardDestination`) to reuse code.